src/macros.jl

```diff
@@ -2161,3 +2161,277 @@ julia> @by df :a begin
 macro by(x, what, args...)
     esc(by_helper(x, what, args...))
 end
+
+##############################################################################
+##
+## @unique - distinct row selection
+##
+##############################################################################
+
+function unique_helper(x, args...)
+    exprs, outer_flags = create_args_vector(args...)
+    t = (fun_to_vec(ex; no_dest=true, outer_flags=outer_flags) for ex in exprs)
+    quote
+        $DataFrames.unique($x, $(t...))
+    end
+end
+
+"""
+    @unique(x, args)
+
+Select unique rows in `AbstractDataFrame`s.
+
+### Arguments
+
+* `d` : an AbstractDataFrame
+* `args` :  keyword-like arguments, of the form `f(:x)` specifying
+symbols to specify identify columns
+
+### Returns
+
+* `::AbstractDataFrame`
+
+Inputs to `@unique` can come in two formats: a `begin ... end` block, or as a series of
+arguments and keyword-like arguments. For example, the following are
+equivalent:
+
+```julia
+@unique df begin 
+    :x + :y
+end
+```
+
+and 
+
+```
+@unique df :x + :y
+```
+
+`@unique` uses the syntax `@byrow` to wrap transformations in
+the `ByRow` function wrapper from DataFrames, apply a function row-wise,
+similar to broadcasting. For example, the call
+
+```
+@unique(df, @byrow :x + :y)
+```
+
+becomes
+
+```
+unique(df, :x => ByRow((x,y) -> x + y))
+```
+
+a transformation which cannot be conveniently expressed
+using broadcasting.
+
+`@unique` allows `@byrow` at the beginning of a block of selections 
+(i.e. `@byrow begin... end`). The transformation in the block will operate by row.
+
+### Examples
+
+```jldoctest
+julia> using DataFramesMeta;
+
+julia> df = DataFrame(x = 1:10, y = 10:-1:1);
+
+julia> @unique(df, :x + :y)
+1×2 DataFrame
+ Row │ x      y      
+     │ Int64  Int64  
+─────┼───────────────
+   1 │     1      10   
+
+julia> @unique df begin
+            :x + :y
+        end
+1×2 DataFrame
+ Row │ x      y      
+     │ Int64  Int64  
+─────┼───────────────
+   1 │     1      10   
+```
+"""
+macro unique(x, args...)
+    esc(unique_helper(x, args...))
+end
+
+function runique_helper(x, args...)
+    exprs, outer_flags = create_args_vector(args...; wrap_byrow=true)
+    t = (fun_to_vec(ex; no_dest=true, outer_flags=outer_flags) for ex in exprs)
+    quote
+        $DataFrames.unique($x, $(t...))
```

Comment:
@bkamins Looking at the docs for `unique`, the fact that transformations can be passed to `unique` is not explicitly documented. 
Instead of using `unique`, do 

```
df2 = select(df, t...; copycols = false)
rowidxs = (!).(nonunique(df2))
df[rowidxs, :]
```

Reply:
`cols` in `unique` is documented as:

> in selected columns or their transformations

so it explicitly mentions that transformations are allowed. But maybe a better wording could be used (this was written a long time ago, when we have a much less precise vocabulary in DataFrames.jl 😄). If you have a proposal out of hand then please comment.